lib/pbench/server/api/resources/query_apis/metadata_index/namespace_and_rows.py

```diff
@@ -66,29 +48,34 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
         """
         run_id = context["run_id"]
         dataset = context["dataset"]
+        document = self.ES_INTERNAL_INDEX_NAMES.get(json_data["target_document"])
```

Comment:
This should be validated, rather than just taking a `KeyError` if they specify a bad value in the URI.

As we discussed, adding a `target_document` KEYWORD Parameter to the class Schema would be a good way to accomplish that.

Reply:
Dave, your "this" is ambiguous:  are you talking about checking whether `json_data` has a `"target_document"` key?  (Because Nikhil does handle the case where an existing `"target_document"` key corresponds to a bad value....)

I'm certainly not saying that having a better Schema wouldn't be better, but I think that, by virtue of it being a path parameter, `json_data` will always have a `"target_document"` key here (at least, it will, if Nikhil corrects the code `ApiBase._dispatch()`)....but I could be wrong.